### PR TITLE
[stack-pr] Convert stack-pr.py into an installable python package

### PR DESCRIPTION
**Pull request by zbowling**
Refactor the bare scripts into a proper python package with a
pyproject.toml.

Add support for a build backend so stack-pr can submitted to the
Python Package Index and be easily installed with pipx.

Fix some minor import issues and lint warnings using ruff.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+.ipynb_checkpoints
+
+# IPython
+profile_default/
+ipython_config.py
+
+# pyenv
+#   For a library or package, you might want to ignore these files since the code is
+#   intended to run in multiple environments; otherwise, check them in:
+# .python-version
```
Comment (ZolotukhinM) [inline .gitignore:88]: Why is this commented out (and some other similar entries)?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,169 @@
+# Byte-compiled / optimized / DLL files
```
Comment (ZolotukhinM) [inline .gitignore:1]: Did you intentionally add `.gitignore`? If yes, I'd suggest cleaning it up from unnecessary entries before merging.

**Review by ZolotukhinM — APPROVED** [review]
Overall looks good to me, and thanks for adding this! One question: did you test that everything still works?

File: .gitignore (hunk 1 above)
Comment (zbowling) [reply .gitignore:88]: Not sure. it was the default boilerplate standard .gitignore from pdm/poetry.

File: .gitignore (hunk 2 above)
Comment (zbowling) [reply .gitignore:1]: Ack. This was from the boilerplate .gitignore from `pdm init` with some extras added from poetry and pixi. Some of it's best practices/common stuff and seemed harmless but I'll delete the unnecessary stuff.

**Pull request merged**